### DISCUSSION TITLE
Needed migrations from v2 to v3

### DISCUSSION THREAD

**Discussion by tdeekens** [panva/jose#108] [Q&A]
First of, great library! Incredible what it offers!

I was however looking into migrating from v2 to v3 and couldn't find any release notes towards the breaking changes. I assume it's mostly around the new revised promise based API. I also checked the commits but I couldn't deduct needed steps from it.

Are there any recommended steps to take from v2 to v3? 

Thanks in advance.

**Comment by panva** [comment]
The API and modules, how they're obtained, etc. is so different that you can just think of it as a new module and follow the docs.

As in #102 tho i have to mention, the API is a subset, not everything is carried over. For that purpose i'm leaving 2.x as a [supported major](https://github.com/panva/jose#supported-versions) so that you don't have to upgrade unless you want to.
